semantic/semantic.fbs

```diff
@@ -0,0 +1,354 @@
+namespace fbsemantic; 
+
+union Statement {
+    OptionStatement,             
+    BuiltinStatement,            
+    TestStatement,               
+    ExpressionStatement,         
+    ReturnStatement,                
+}
+
+union Assignment {
+    MemberAssignment,            
+    NativeVariableAssignment, 
+}
+
+union Expression  {
+    StringExpression,       
+    ArrayExpression,        
+    FunctionExpression,     
+    BinaryExpression,       
+    CallExpression,         
+    ConditionalExpression,  
+    IdentifierExpression,   
+    LogicalExpression,      
+    MemberExpression,       
+    IndexExpression,        
+    ObjectExpression,       
+    UnaryExpression,        
+    BooleanLiteral,          
+    DateTimeLiteral,         
+    DurationLiteral,         
+    FloatLiteral,            
+    IntegerLiteral,          
+    StringLiteral,           
+    RegexpLiteral,           
+    UnsignedIntegerLiteral,  
+}
+
+union PropertyKey {
+    Identifier,
+    StringLiteral,
+}
+
+table Package {
+    loc:SourceLocation;  
+    package:string; 
+    files:[File];
+}
+
+table File {
+    loc:SourceLocation; 
+    package:PackageClause;
+    imports:[ImportDeclaration];
+    body:[WrappedStatement];
+}
+
+table PackageClause {
+    loc:SourceLocation;
+    name:Identifier;
+}
+
+table ImportDeclaration {
+    loc:SourceLocation; 
+    as:Identifier;
+    path:StringLiteral;
+}
+
+table Block {
+    loc:SourceLocation;
+    body:[WrappedStatement];
+}
+
+table SourceLocation {
+    file:string;
+    start:Position;
+    end:Position;
+    source:string;
+}
+
+struct Position {
+    line:int;
+    column:int;
+}
+
+// Statements
+
+// Unions cannot be vector elements,
+// so wrap the union in a table.
+table WrappedStatement {
+    statement:Statement;
+}
+
+table OptionStatement {
+    loc:SourceLocation;
+    assignment:Assignment;
+}
+
+table BuiltinStatement {
+    loc:SourceLocation; 
+    id:Identifier; 
+}
+
+table TestStatement {
+    loc:SourceLocation; 
+    assignment:NativeVariableAssignment;
+}
+
+table ExpressionStatement {
+    loc:SourceLocation; 
+    expression:Expression; 
+}
+
+table ReturnStatement {
+    loc:SourceLocation; 
+    argument:Expression; 
+}
+
+table NativeVariableAssignment {
+    loc:SourceLocation;
+    identifier:Identifier; 
+    init_:Expression; 
+}
+
+table MemberAssignment {
+    loc:SourceLocation; 
+    member:MemberExpression;
+    init_:Expression; 
+}
+
+table Extern {
+    loc:SourceLocation; 
+    assignments:[ExternalVariableAssignment];
+    block:ExternBlock; 
+}
+
+table ExternalVariableAssignment {
+    loc:SourceLocation; 
+    identifier:Identifier;
+}
+
+table ExternBlock {
+    loc:SourceLocation;
+    node:Block;
+}
```

Comment:
These are no longer concepts in the Rust semantic graph.

Reply:
Ahh, okay. I'll go ahead and remove these.